### PR TITLE
Update to Z3 4.8.7

### PR DESCRIPTION
**Pull request by leonardoalt**
1- @ekpyron can you update the PPA?
2- After that we'd need someone to build and upload the new docker.
3- Then we can retrigger the tests here.

### CODE REVIEW

**Comment by ekpyron** [comment]
I'll update the PPA. Would you have permissions to do that yourself? Just to avoid such permission-nightmares as we had for the docker stuff ;-)...

**Comment by leonardoalt** [comment]
I don't think I have permissions for any of those.

**Comment by ekpyron** [comment]
We should probably change that... I just uploaded new z3 ppa package sources, I'll post links to the builds once they are accepted and started.

**Comment by ekpyron** [comment]
Builds are running here: https://launchpad.net/~ethereum/+archive/ubuntu/cpp-build-deps/+packages

Can you change ``4.8.6`` to ``4.8.7`` in ``scripts/deps-ppa/static_z3.sh`` in this PR?

**Comment by leonardoalt** [comment]
Will do

**Comment by bshastry** [comment]
I think the clang based test CI failure is because the clang docker image still points to z3-4.8.6. I think changing the z3 version in clang docker file to `z3-4.8.7` and updated the dockerfile/ image/circle ci docker image revision might be sufficient to fix it

- Change this line to `z3-4.8.7`
https://github.com/ethereum/solidity/blob/9554b90e07895eb80c44f0a01cc26e19fbf3a5fc/.circleci/docker/Dockerfile.clang.ubuntu1904#L65

- Change this line to `3`
https://github.com/ethereum/solidity/blob/9554b90e07895eb80c44f0a01cc26e19fbf3a5fc/.circleci/config.yml#L12

- Tag all docker images as <name>-3 and push to docker hub
  - Note that although the ubuntu1804 and 1904 docker images have not changed (They fetch z3 lib from PPA I believe), we will need to retag and repush them because of the way circle ci docker revisions have been configured
  - To fix this, we would need two additional pipeline parameters called something like `ubuntu1804-revision` and `ubuntu1904-revision` that tracks revisions of the respective ubuntu docker images

**Comment by leonardoalt** [comment]
Yea we need the PPA to build first, which now depends on https://github.com/Z3Prover/z3/issues/2776

**Comment by leonardoalt** [comment]
Depends on https://github.com/ethereum/solidity/pull/7879

File: .circleci/docker/Dockerfile.clang.ubuntu1904 [hunk 1]
```diff
@@ -62,7 +62,7 @@ RUN git clone --recursive -b boost-1.69.0 https://github.com/boostorg/boost.git
     rm -rf /usr/src/boost
 
 # Z3
-RUN git clone --depth 1 -b z3-4.8.6 https://github.com/Z3Prover/z3.git \
+RUN git clone --depth 1 -b z3-4.8.7 https://github.com/Z3Prover/z3.git \
```
Comment (leonardoalt) [inline .circleci/docker/Dockerfile.clang.ubuntu1904:65]: @ekpyron how would it work to update it here though? Wouldn't this need the patch as well?

Comment (leonardoalt) [reply .circleci/docker/Dockerfile.clang.ubuntu1904:65]: Ah no, this is 64bit

Comment (ekpyron) [reply .circleci/docker/Dockerfile.clang.ubuntu1904:65]: I hope at least, it should work, yes. If not we'll need to remove ``--depth 1`` and cherry-pick the commit here as well, but let's hope that's not necessary.

**Comment by leonardoalt** [comment]
One more test that has different behavior locally vs CI...
This should change with https://github.com/ethereum/solidity/pull/7873

**Review by chriseth — APPROVED** [review]

**Pull request merged**